### DISCUSSION TITLE
Disable firebase initialization in China region

### DISCUSSION THREAD

**Discussion by tomcheung1208** [invertase/react-native-firebase#5205] [General]
I would like to know how to disable the firebase initialization When user was in China. I have a app and It will take a long time and enter the first page. Around 30 seconds. I saw the log show multiple address (firebaseinstallations.googleapis.com) after the app started. I think these calling is the reason. I searched  but I still can't find the solution.
Thanks

**Comment by mikehardy** [comment]
There is not enough information in your question to say anything definitive. Perhaps if you are using one of the modules that has a "GDPR"-like flow where you may delay collection of information you could attempt to turn off the automatic collection of information and it would work?

**Comment by tomcheung1208** [comment]
![3E6F2C38-AE4E-4054-9954-9A654428B9D0](https://user-images.githubusercontent.com/50606721/115977078-f2b47900-a5a6-11eb-99f2-1d27a1678a99.png)
I tried to create a firebase.json and add "analytics_auto_collection_enabled": false, but it seem doesn't work.
I don't have set any special setup, just use RNFirebase.app().analytics() in some places. Also I have call a api to check the region whether is China before use firebase

**Reply by mikehardy** [reply]
I think you're going to have to take this up with upstream native SDK team - assuming and provided you reproduce it with their analytics quickstart (which will hopefully make it easy...) 

You gave such a small amount of information to reason about here I'm not sure if it's even android or ios so I'll link you to the Android quickstart from firebase-android-sdk team - https://github.com/firebase/quickstart-android/tree/master/analytics
